aepsych/server/sockets.py

```diff
@@ -115,65 +116,35 @@ def accept_client(self):
                     try:
                         self.conn, self.addr = sock.accept()
                         logger.info(
-                            f"Connected by {self.addr}, waiting for messages...")
+                            f"Connected by {self.addr}, waiting for messages..."
+                        )
                         client_not_connected = False
                     except Exception as e:
-                        logger.info(
-                            f'Connection to client failed with error {e}')
+                        logger.info(f"Connection to client failed with error {e}")
+                        raise Exception
 
     def receive(self, server_exiting):
 
-        # Set up the logic for reading using select.
-        # Setting the timeout to 0 makes it nonblocking.
-        # Magic Number tweaking can be made with the timeout to cut down CPU Usages vs User Experience
+        # Using the select module allows you to pass a collection of sockets that the filesystem is scanning for changes.
+        # Select takes 3 lists as input, lists that contain sockets to poll. These polled sockets will be checked to see if they are readable,writeable,or contain an error.
+        # It also takes in 1 optional input, the timeout.
+        # setting the timeout to 0 makes it constantly poll the sockets for their status. In this case, the socket is being polled constantly to see if anything was written to it.
+        # this is expensive and not best behavior, but it makes for a more seamless experience, as if there is a timeout, the socket will actively block, which makes it difficult
+        # to scan for serverside instructions, such as a shutdown message.
 
         while not server_exiting:
             rlist, wlist, xlist = select.select([self.conn], [], [], 0)
             for sock in rlist:
                 try:
                     if rlist:
-                        recv_result = sock.recv(1024*512)
+                        recv_result = sock.recv(1024 * 512)
```

Comment:
So 1024 * 512 is the timeout parameter? How was this number chosen?

Reply:
No, 1024 * 512 is the max size of the message that the socket can receive at once.

Reply:
the timeout is 0, it is the value at the end of select.select().

The parameter for select.select are 3 lists and an optional timeout.
The three lists should contain sockets that you are listening to. the first list lets you poll for sockets that can be read, the second to sockets that can be written to, and the third, for sockets you want to monitor for errors.

 I put a 0 timeout, which turns the select.select method from a blocking method (if there is a timeout, it halt the program until it sees a change in a socket) to a non-blocking method. This means that it is constantly polling the sockets for changes, and simply does nothing if it sees that nothing was done to the socket.